### PR TITLE
Makefile tweaks and Nix{,OS} support

### PR DESCRIPTION
**Pull request by vs49688**
These changes allow openrussian-cli to be used on Nix and NixOS.

Also fixes the database download URL.

Tested on my own system.

### CODE REVIEW

**Comment by rhaberkorn** [comment]
Thanks! I did not know that anyway is actually using this (besides myself).

A few questions:
* Will this be uploaded to any official NixOS repositories?
* What if we add packaging for some other distribution (e.g. Debian)? We would certainly add a subdirectory. Perhaps the Nix/OS files should be in their own subdirectory (`nix/`) as well?
* Why do you check for the existence of Lua, but not any of the other tools required (e.g. AWK, sqlite3)? These kinds of checks are quite uncommon in Makefiles. I suggest to remove them. If we need any serious configure-time checks, I'd rather add an Autoconf/Autotools build system.
* You appear to have checked/tested a recent database dump? If so, you can update `openrussian-sql.zip`.

**Comment by rhaberkorn** [comment]
If this will be upstreamed to NixOS, it would also be nice to add a few remarks to README on how to install it.

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  postInstall = ''
+    installShellCompletion --bash --name openrussian ./openrussian-completion.bash
+    installManPage ./openrussian.1
+  '';
+
+  meta = with lib; {
+    homepage    = "https://github.com/rhaberkorn/openrussian-cli";
+    description = "Offline Console Russian Dictionary (based on openrussian.org)";
+    license     = licenses.gpl3;
```
Comment (rhaberkorn) [inline default.nix]: Licensing is a bit more complex - see "License" in README. But I don't know how this is handled in Nix. On Debian you could specify the license of every file individually.

**Comment by vs49688** [comment]
> * Thanks! I did not know that anyway is actually using this (besides myself).

It's a recent thing. I only discovered it a few days ago!

> * Will this be uploaded to any official NixOS repositories?
> * What if we add packaging for some other distribution (e.g. Debian)? We would certainly add a subdirectory. Perhaps the Nix/OS files should be in their own subdirectory (nix/) as well?

Это план. Once it is, the `default.nix` and `shell.nix` will no longer be needed. I just committed them so they had somewhere to go in the meantime.

> * Why do you check for the existence of Lua, but not any of the other tools required (e.g. AWK, sqlite3)? These kinds of checks are quite uncommon in Makefiles. I suggest to remove them. If we need any serious configure-time checks, I'd rather add an Autoconf/Autotools build system.

Ah, that's because I had some early build issues with Nix. I'll remove them.

> * You appear to have checked/tested a recent database dump? If so, you can update openrussian-sql.zip.

...I thought I did. I'll fix it.

> If this will be upstreamed to NixOS, it would also be nice to add a few remarks to README on how to install it.

Once it's upstream'd, absolutely!

File: default.nix (hunk 1 above)
Comment (vs49688) [reply default.nix]: You can specify multiple licenses. See https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes

Fixed in https://github.com/rhaberkorn/openrussian-cli/commit/732172e2b49a1c468d9097cf937d8da83e6f061f.

**Comment by rhaberkorn** [comment]
Благодарю. So once this has been upstreamed, the port will live somewhere else and the `*.nix` files can be removed? I like this since I cannot maintain this port. (I'm usually on Ubuntu.)

**Comment by vs49688** [comment]
That's right! Soon (probably tomorrow) I'll submit a PR to https://github.com/NixOS/nixpkgs, and the `*.nix` files can be removed. I'll add something to the README afterwards too.

**Comment by vs49688** [comment]
Oh, that reminds me. Could you please create a release tag with a version number? I've just put 1.0.0 in `default.nix` as a placeholder. When I upstream it, it should reference something that has a version.

**Comment by rhaberkorn** [comment]
Turns out that `make check` did *not* run through without errors. I've fixed this and created the tag as you wished.

**Comment by vs49688** [comment]
PR submitted: https://github.com/NixOS/nixpkgs/pull/120012

**Comment by vs49688** [comment]
PR accepted. When it hits the `unstable` channel (should be in a few days), I'll send a new PR removing the files and adding a note in the README.

**Pull request merged**